#! /usr/bin/env python

PACKAGE='hd_driver'
import roslib; roslib.load_manifest(PACKAGE)

from device_driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

#       Name                   Type      Reconfiguration level             Description         Default   Min   Max
gen.add("port",                str_t,    SensorLevels.RECONFIGURE_STOPPED, "The port threw which to talk to the device", "/dev/ttyUSB2")

gen.add("control_service",     str_t,    SensorLevels.RECONFIGURE_OPEN, "The name of the set position action server", "hd_control")
gen.add("info_topic",          str_t,    SensorLevels.RECONFIGURE_OPEN, "The topic on which motor info is published", "roboteq_info")
gen.add("pause_topic",         str_t,    SensorLevels.RECONFIGURE_OPEN, "The topic on which the pause message is listened for", "/pause")

gen.add("reference_frame",     str_t,    SensorLevels.RECONFIGURE_OPEN, "The reference frame for motor info", "/hd_reference_frame")

gen.add("feedback_rate",       double_t,    SensorLevels.RECONFIGURE_OPEN, "The rate at which feedback is published", 1, 0.1, 10)


exit(gen.generate(PACKAGE, "hd_driver", "HD"))
